### PR TITLE
Add more swizzle choices

### PR DESCRIPTION
**Pull request by Disiuze**
Based on #1779 and #2009, this PR adds 18 new swizzle options.
 
`swizzle 27`
image

`swizzle 20`
image

### CODE REVIEW

**Comment by ghoulavenger** [comment]
Not sure I care about the overlays, but I'd like to change regular colors, since doing the FW campaign pretty much locks you into green (I miss the blue merchant swizzle). The only question I have is did you include the ability to lock the swizzle during the FW campaign (because at that point you really need to be wearing their colors, it's a lore thing).

**Comment by Amazinite** [comment]
https://github.com/endless-sky/endless-sky/issues/1779#issuecomment-266922948

>I don't think it makes sense to have 0-6 be normal swizzles, then 7 and 8 be strange, then it returns to normal at 9-28.

Same story here.

File: source/SpriteShader.cpp [hunk 1]
```diff
@@ -31,20 +31,43 @@ namespace {
 	GLint blurI;
 	GLint clipI;
 	GLint alphaI;
-	
+
```
Comment (tehhowch) [inline source/SpriteShader.cpp]: Your editor is eating whitespacing.

**Comment by Disiuze** [comment]
> The only question I have is did you include the ability to lock the swizzle during the FW campaign (because at that point you really need to be wearing their colors, it's a lore thing).

What do you mean by this? This PR only adds new swizzles.
>Same story here. I also question the purpose of the overlays.

I'd like to keep it that way as I do not want to cause problems with cloaking or outlines.
What the overlays are used for is up to the content creators. I could see them being used for an alien government or unique NPC or ship.

**Comment by ghoulavenger** [comment]
> What do you mean by this? This PR only adds new swizzles.

Ah, I was hoping for something that branches off of it then. I'd like to choose my own swizzle and read that as part of the PR (mistakenly).

**Comment by Amazinite** [comment]
Given the fact that swizzles don't apply to a ship's turrets, I don't see the overlays being necessary in this PR until that happens. The picture you have for swizzle 31 shows what I mean perfectly; the turrets just pop out at you an look completely wrong on the ship.

**Comment by czartrak** [comment]
I found myself wanting a black swizzle that doesnt make my entire ship black while writing a mission yesterday, so i'd like to have these

**Comment by Lineth** [comment]
More swizzles would always be great. Quarg/Coalition would look good in black (probably the coalition more).

**Comment by endless-sky** [comment]
As I've said before when this has come up: ships just don't look good if their color palette is limited to a 1D or 2D slice of the full 3D RGB color space. If it were possible to apply an arbitrary *hue shift* to the ship, so the colors changed while still occupying a full 3D color space, that would be a different story:

hue shift

That could be done by multiplying every pixel's RGB value with a 3x3 color transformation matrix, so it's doable and not too complicated. I'm not sure how much GPU overhead it would add, though.

**Comment by Zitchas** [comment]
I'm not a programmer, but could you clarify in layman's terms how much overhead this would add? I'm running on a 6 year old desktop with a 2+ year old GPU, and ES currently barely registers as using any resources at all. I appreciate keeping the requirements low, but I think having a variety of new swizzles would be worth the GPU overhead (unless we're talking "becomes taxing for a modern GPU" level of workload, anyway)

**Comment by Zitchas** [comment]
Is anything being done in terms of new swizzles or overlays?

I still really think we should have more than the current six, and overlays look like a good addition to me:
a) I'm getting sick of FW green
b) I suspect that I will get equally sick of Republic or Syndicate colours once getting locked into those.
c) As a content creator, having alternate swizzles for other alien races that aren't duplicating already-existing ones would be much appreciated.
